llvm/test/CodeGen/AMDGPU/waterfall_kills_scc.ll

```diff
@@ -0,0 +1,87 @@
+; NOTE: Assertions have been autogenerated by utils/update_llc_test_checks.py UTC_ARGS: --version 3
+; RUN:  llc -march=amdgcn -mcpu=gfx906 -verify-machineinstrs < %s | FileCheck -check-prefix=GFX906 %s
```

Comment:
I'm not sure I like a IR test alone for this kind of fix. I would rather see a MIR test with `-run-pass=si-fix-sgpr-copies". Is that doable?
A MIR test can probably be smaller as well, and test more edge cases.

The IR test can stay too - It's just that I would like a more focused MIR test.
For the IR test I would also check more than one arch. Not sure which ones are relevant in this case but I think at least GFX8/10/11 would be nice to see. Perhaps gfx90a too. 

In any case I also think at least one wave32/wave64 test is needed (for both the MIR and IR tests)

Reply:
I added two mir  tests, one for 906 (wave64) and one for 1030 (wave32 in the new patch.
I also removed the larger IR test. I will put that test as a part of another patch (https://github.com/llvm/llvm-project/pull/68363) once this is committed.